Framework/API/src/FileFinder.cpp

```diff
@@ -533,6 +543,20 @@ void FileFinderImpl::getUniqueExtensions(
   }
 }
 
+/**
+ * Performs validation on the search text entered into the File Finder. It will
+ * throw an exception if a problem is found. This error would then be caught and
+ * displayed to a user.
+ * @param searchText :: The text to validate.
+ * @throw std::invalid_argument if the searchText is invalid
+ */
+void FileFinderImpl::validateRuns(const std::string &searchText) const {
```

Comment:
Thanks for this. Could I ask why we prefer an exception rather than say returning a `std::string` containing any possible error message? Using an exception indicates that it's not expected that the text might be incorrect but here both answers of is valid or not are valid answers and not exceptional states.

Reply:
I used a `throw` here because the `findRuns` method doesn't have a convenient way to return an error message. Instead it throws whenever there is a problem and the `FindFilesWorker` creates the error message stuff. I am not sure how to change this without making a large change to `findRuns`

Reply:
Ah I see - I didn't spot that below. In this case I would then have `findRuns` throw if this function returns it's not valid rather than forcing the user of the validation to use an exception if all they want to do is check that the string is valid input.

Reply:
Ok sounds good 👍